File: LeetCode/problems/48-rotate-image/48-rotate-image.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var rotate = function(matrix) {
    for(let R = 0; R<matrix.length; R++){
        for(let C = R+1; C<matrix.length; C++){
            [matrix[R][C],matrix[C][R]] = [matrix[C][R],matrix[R][C]];
        }
    }

    matrix.map( (row)=> row.reverse() );
};

File: LeetCode/problems/48-rotate-image/48-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        for R in range(0,len(matrix)):
            for C in range(R+1,len(matrix)):
                matrix[R][C], matrix[C][R] = matrix[C][R], matrix[R][C]
        
        for R in matrix:
            R.reverse()
